spec/rototiller/utilities/command_flag_spec.rb

```diff
@@ -105,9 +122,116 @@
+        let(:args) { {:name => flag, :is_boolean => true, :override_env => override_env} }
+
+        it 'should report the switch name' do
+          expect(subject.flag).to eq(flag)
+        end
+        it 'should message with information about flag' do
+          expected_message = /The CLI switch '#{flag}' will be used./
+          expect(subject.message).to match(expected_message)
+        end
+        it 'should not have stop set to true' do
+          expect(subject.stop).to be_falsey
+        end
+      end
+
+      context 'should set switch to :override_env with no default and env_override passed and set' do
+        let(:args) { {:name => flag, :is_boolean => true, :override_env => override_env} }
+
+        it 'should report the switch name' do
+          ENV[override_env] = value
+          expect(subject.flag).to eq(value)
+        end
+
+        # FIXME: repeated ENV setting here is problematic
+        #   doesn't work in before nor let.
+        it 'should message with information about flag' do
+          ENV[override_env] = value
+          expected_message = /The CLI switch '#{value}' will be used./
+          expect(subject.message).to match(expected_message)
+        end
+        it 'should not have stop set to true' do
+          expect(subject.stop).to be_falsey
+        end
+      end
+
+      context 'should set switch to :default over :name when env is not set so it can default to "off"' do
+        let(:args) { {:name => flag, :is_boolean => true, :default => value} }
+
+        it 'should report the switch name' do
+          expect(subject.flag).to eq(value)
+        end
+        it 'should message with information about flag' do
+          expected_message = /The CLI switch '#{value}' will be used./
+          expect(subject.message).to match(expected_message)
+        end
+        it 'should not have stop set to true' do
+          expect(subject.stop).to be_falsey
+        end
+      end
+
+      context 'should set switch to :default of ""' do
+        let(:args) { {:name => flag, :is_boolean => true, :default => ''} }
+
+        it 'should report the switch name' do
+          expect(subject.flag).to eq('')
+        end
+        it 'should message with information about flag' do
+          expected_message = /The CLI switch '#{flag}' will NOT be used./
+          expect(subject.message).to match(expected_message)
+        end
+        it 'should not have stop set to true' do
+          expect(subject.stop).to be_falsey
+        end
+      end
+
+      context 'should set switch to env_override when env is set with a :default' do
+        let(:args) { {:name => flag, :is_boolean => true, :default => value, :override_env => override_env} }
+        let(:env_val) { "env_#{value}" }
+
+        it 'should report the switch name' do
+          ENV[override_env] = env_val
+          expect(subject.flag).to eq(env_val)
+        end
+
+        # FIXME: repeated ENV setting here is problematic
```

Comment:
I don't understand what is problematic about this.  Did you want to set the ENV in a before?